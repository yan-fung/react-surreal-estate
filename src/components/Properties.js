import React from "react";
import "../styles/properties.css";

const Properties = () => {
  return (
    <div className="property_header">
      <h1>Properties Page</h1>
    </div>
  );
};

export default Properties;
