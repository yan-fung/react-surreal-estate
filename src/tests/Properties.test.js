import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Properties from "../components/Properties";

describe("Properties", () => {
  const validProps = {
    properties: [
      {
        key: "1234",
        title: "Two beds house",
        bathrooms: "2",
        bedrooms: "2",
        type: "Semi-detached",
        price: "1000",
        email: "testing@gmail.com",
        city: "Leeds",
      },
      {
        key: "1234",
        title: "Two beds house",
        bathrooms: "2",
        bedrooms: "2",
        type: "Semi-detached",
        price: "1000",
        email: "testing@gmail.com",
        city: "Leeds",
      },
    ],
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Properties properties={validProps.properties} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  // it("renders correct values for props", () => {
  //   const { getByText } = render(
  //     <Properties properties={validProps.properties} />
  //   );
  //   expect(getByText("Two beds house")).toHaveClass("item");
  // });
});
