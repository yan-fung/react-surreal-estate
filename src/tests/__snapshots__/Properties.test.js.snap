// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property_header"
  >
    <div
      class="property_container"
    >
      <div
        class="sidebar"
      >
        <div
          class="sidebar"
        >
          <form
            class="search-box"
          >
            <input
              class="search-input"
              placeholder="Search"
              type="text"
              value=""
            />
            <button
              class="button"
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-magnifying-glass icon-search"
                data-icon="magnifying-glass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </form>
          <div
            class="sidebar_city"
          >
            <h3>
              Filter by City
            </h3>
            <ul>
              <li>
                <a
                  href="/?query={\\"city\\":\\"Manchester\\"}"
                >
                  Manchester
                </a>
              </li>
            </ul>
            <ul>
              <li>
                <a
                  href="/?query={\\"city\\":\\"Leeds\\"}"
                >
                  Leeds
                </a>
              </li>
            </ul>
            <ul>
              <li>
                <a
                  href="/?query={\\"city\\":\\"Sheffield\\"}"
                >
                  Sheffield
                </a>
              </li>
            </ul>
            <ul>
              <li>
                <a
                  href="/?query={\\"city\\":\\"Liverpool\\"}"
                >
                  Liverpool
                </a>
              </li>
            </ul>
          </div>
          <div
            class="sidebar_sort"
          >
            <h3>
              Sort by
            </h3>
            <ul>
              <li>
                <a
                  href="/?sort={\\"price\\":1}"
                >
                  Price ascending
                </a>
              </li>
            </ul>
            <ul>
              <li>
                <a
                  href="/?sort={\\"price\\":-1}"
                >
                  Price descending
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="properties"
      />
    </div>
  </div>
</DocumentFragment>
`;
